src/crimson/os/seastore/transaction_manager.h

```diff
@@ -175,10 +175,30 @@ class TransactionManager : public ExtentCallbackInterface {
   base_iertr::future<TCachedExtentRef<T>> read_pin(
     Transaction &t,
     LBAMappingRef pin)
+  {
+    return seastar::do_with(
+      std::move(pin),
+      [this, &t](auto &pin) {
+      return get_extent_if_linked<T>(t, pin
+      ).si_then([this, &t, &pin](auto extent) {
+	if (extent) {
+	  return base_iertr::make_ready_future<
+	    TCachedExtentRef<T>>(extent);
+	} else {
+	  return pin_to_extent<T>(t, std::move(pin));
```

Comment:
IIUC `pin_to_extent()` should be called atomically with `v.has_child()` (specifically  `FixedKVNode::get_child(...).has_child()`) -- the absent extent should be created&linked atomically in the lba tree upon the read attempt.

This is because extent reading might be issued concurrently, if check-child and create&link-child-extent are not atomic, the concurrent readers might end up with creating independent extents and link them independently as the same child.

It is lucky to identify this time, but might not in the future, which can cause confusing errors. If it's not straitforward to encapsulate the entire atomic process, probably worth to add comments to highlight.

Reply:
Ah, yes, will try to solve it.

Reply:
Just fixed it.